EnemyAI.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using Pathfinding;
+using UnityEngine;
+
+
+public class EnemyAI : MonoBehaviour
+{
+
+    public Transform target;
+
+    public float speed = 200f;
+    public float nextWaypointDistance = 3f;
+
+    private Path path;
+    private int currentWaypoint = 0;
+    private bool reachedEndOfPath = false;
+
+    private Seeker seeker;
+    private Rigidbody2D rb;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        seeker = GetComponent<Seeker>();
+        rb = GetComponent<Rigidbody2D>();
+
+        InvokeRepeating("UpdatePath", 0f, .5f);
+    }
+
+    private void UpdatePath()
+    {
+        if (seeker.IsDone())
+        {
+            seeker.StartPath(rb.position, target.position, OnPathComplete);
+        }
+    }
+
+    private void OnPathComplete(Path p)
+    {
+        if (p.error) return;
+        path = p;
+        currentWaypoint = 0;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (path == null)
+            return;
+
+        if (currentWaypoint >= path.vectorPath.Count)
+        {
+            reachedEndOfPath = true;
+            return;
+        }
+        else
+        {
+            reachedEndOfPath = false;
+        }
+
+        var direction = ((Vector2)path.vectorPath[currentWaypoint] - rb.position).normalized;
+        var force = direction * speed * Time.deltaTime;
```

Comment:
When a variable is used in only one place and thereafter it is not used, I recommend not declaring it.
See my suggestion.

```suggestion
        var force = (((Vector2)path.vectorPath[currentWaypoint] - rb.position).normalized) * speed * Time.deltaTime;
```